drivers/ps2/ps2_npcx_controller.c

```diff
@@ -0,0 +1,372 @@
+		if (status & BIT(NPCX_PSTAT_RFERR))
+			LOG_ERR("RX Frame error");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static void ps2_npcx_ctrl_isr(const struct device *dev)
+{
+	uint8_t active_ch, mask;
+	struct ps2_reg *const inst = HAL_PS2_INSTANCE(dev);
+	struct ps2_npcx_ctrl_data *const data = DRV_DATA(dev);
+
+	/*
+	 * ACH = 1 : Channel 0
+	 * ACH = 2 : Channel 1
+	 * ACH = 4 : Channel 2
+	 * ACH = 5 : Channel 3
+	 */
+	active_ch = GET_FIELD(inst->PSTAT, NPCX_PSTAT_ACH);
+	active_ch = active_ch > 2 ? (active_ch - 2) : (active_ch - 1);
+	LOG_DBG("ACH: %d\n", active_ch);
+
+	/*
+	 * Inhibit PS/2 transaction of the other non-active channels by
+	 * pulling down the clock signal
+	 */
+	mask = ~NPCX_PSOSIG_CLK_MASK_ALL | BIT(NPCX_PSOSIG_CLK(active_ch));
+	inst->PSOSIG &= mask;
+
+	/* PS/2 Start of Transaction */
+	if (IS_BIT_SET(inst->PSTAT, NPCX_PSTAT_SOT) &&
+	    IS_BIT_SET(inst->PSIEN, NPCX_PSIEN_SOTIE)) {
+		/*
+		 * Once set, SOT is not cleared until the shift mechanism
+		 * is reset. Therefore, SOTIE should be cleared on the
+		 * first occurrence of an SOT interrupt.
+		 */
+		inst->PSIEN &= ~BIT(NPCX_PSIEN_SOTIE);
+		LOG_DBG("SOT");
+		/* PS/2 End of Transaction */
+	} else if (IS_BIT_SET(inst->PSTAT, NPCX_PSTAT_EOT)) {
+		inst->PSIEN &= ~BIT(NPCX_PSIEN_EOTIE);
+
+		/*
+		 * Clear the CLK of active channel to reset
+		 * the shift mechanism
+		 */
+		inst->PSOSIG &= ~BIT(NPCX_PSOSIG_CLK(active_ch));
+
+		/* Tx is done */
+		if (IS_BIT_SET(inst->PSCON, NPCX_PSCON_XMT)) {
+			/* Change the PS/2 module to receive mode */
+			inst->PSCON &= ~BIT(NPCX_PSCON_XMT);
+			k_sem_give(&data->tx_sync_sem);
+		} else {
+			if (ps2_npcx_ctrl_is_rx_error(dev) == 0) {
+				ps2_callback_t callback;
+				uint8_t data_in = inst->PSDAT;
+
+				LOG_DBG("Recv:0x%02x", data_in);
+				callback = data->callback_isr[active_ch];
+				if (callback != NULL)
+					callback(dev, data_in);
+			}
+		}
+
+		/* Restore the enabled channel */
+		inst->PSOSIG |= data->channel_enabled_mask;
+		/*
+		 * Re-enable the Start Of Transaction interrupt when
+		 * the shift mechanism is reset
+		 */
+		inst->PSIEN |= BIT(NPCX_PSIEN_SOTIE);
+		inst->PSIEN |= BIT(NPCX_PSIEN_EOTIE);
+		LOG_DBG("EOT");
+	}
+}
+
+/* PS/2 driver registration */
+static int ps2_npcx_ctrl_init(const struct device *dev);
+
+static struct ps2_npcx_ctrl_data ps2_npcx_ctrl_data_0;
+
+static const struct ps2_npcx_ctrl_config ps2_npcx_ctrl_config_0 = {
+	.base = DT_INST_REG_ADDR(0),
+	.clk_cfg = NPCX_DT_CLK_CFG_ITEM(0),
+};
+
+DEVICE_DT_INST_DEFINE(0, &ps2_npcx_ctrl_init, NULL, &ps2_npcx_ctrl_data_0,
+		      &ps2_npcx_ctrl_config_0, POST_KERNEL,
+		      CONFIG_PS2_INIT_PRIORITY, NULL);
+
+static int ps2_npcx_ctrl_init(const struct device *dev)
+{
+	const struct ps2_npcx_ctrl_config *const config = DRV_CONFIG(dev);
+	struct ps2_npcx_ctrl_data *const data = DRV_DATA(dev);
+	struct ps2_reg *const inst = HAL_PS2_INSTANCE(dev);
+	const struct device *clk_dev = device_get_binding(NPCX_CLK_CTRL_NAME);
```

Comment:
#36557 just merged. Here should modify after rebase. `DEVICE_DT_GET(NPCX_CLK_CTRL_NODE);`

Reply:
Done.